src/main/include/utils/Commands/DriveCommands.h

```diff
@@ -25,50 +26,91 @@ namespace DrivingCommands {
+  auto& fixtureLocations = side == frc::DriverStation::Alliance::kRed
+                               ? Locations::RedFixtureLocations
+                               : Locations::BlueFixtureLocations;
+  return fixtureLocations;
+}
+
+static std::vector<RelativeLocation> GetDistancesToFixtures(
+    frc::Pose2d currentPose) {
+  std::vector<RelativeLocation> locationDistances;
+  auto fixtureLocations = GetFixtureLocations();
+  locationDistances.reserve(fixtureLocations.size());
+  std::transform(fixtureLocations.begin(), fixtureLocations.end(),
                  std::back_inserter(locationDistances),
                  [currentPose](const Locations::FixtureLocation& loc) {
-                   auto dif = currentPose - loc.location;
+                   auto dif = currentPose - loc.fixtureLocation;
                    auto distance = std::hypot(dif.X().value(), dif.Y().value());
                    return RelativeLocation{
                        .hypotDistance = units::meter_t(distance),
-                       .desiredRotation = loc.desiredRotation};
+                       .trackedPose = loc.trackedPose,
+                       .fixtureLocation = loc.fixtureLocation,
+                       .locationRadius = loc.locationRadius};
                  });
+
+  return locationDistances;
+}
+
+static std::vector<RelativeLocation> GetSortedLocations(
+    frc::Pose2d currentPose) {
+  auto locationDistances = GetDistancesToFixtures(currentPose);
+
+  std::sort(locationDistances.begin(), locationDistances.end(),
+            [](const RelativeLocation& a, const RelativeLocation& b) {
+              return a.hypotDistance < b.hypotDistance;
+            });
+
+  return locationDistances;
+}
+
+static RelativeLocation GetClosestFixture(
+    std::vector<RelativeLocation>& locationDistances) {
   auto it = std::min_element(
       locationDistances.begin(), locationDistances.end(),
       [](const RelativeLocation& a, const RelativeLocation& b) {
         return a.hypotDistance < b.hypotDistance;
       });
-  return it->desiredRotation;
+
+  return *it;
 }
 
-static frc2::CommandPtr SnapToAngle(DriveSubsystem* drive) {
-  return (ConsoleInfo("SnapToAngle", "Snapping to a new angle%s", "")
-              .AndThen(TurnToAngle(
-                           drive,
-                           [drive] { return RotationFromProximity(drive); },
-                           false)
-                           .ToPtr()))
-      .WithTimeout(2_s)
-      .FinallyDo([drive] {
-        frc::ChassisSpeeds chassisSpeeds = {
-            .vx = 0_mps, .vy = 0_mps, .omega = 0_rad_per_s};
-        drive->Drive(chassisSpeeds);
-      });
+static RelativeLocation LocationFromProximity(frc::Pose2d currentPose) {
+  auto locationDistances = GetDistancesToFixtures(currentPose);
+
+  auto closestFixture = GetClosestFixture(locationDistances);
+  return closestFixture;
 }
+
+// static frc2::CommandPtr SnapToAngle(DriveSubsystem* drive) {
```

Comment:
Commented out or removed? Should we trim this before merge?

Reply:
yeah, I can do that